onnxruntime/core/providers/op_kernel_type_control_overrides.inc

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace onnxruntime {
+namespace op_kernel_type_control {
+
+// specify allowed types here
+
+// Examples:
+// Specify allowed types per Op kernel arg:
+//   ORT_SPECIFY_OP_KERNEL_ARG_ALLOWED_TYPES(Cast, Input, 0, float, int64_t);
+//   ORT_SPECIFY_OP_KERNEL_ARG_ALLOWED_TYPES(Cast, Output, 0, float, int64_t);
+// Specify allowed types globally:
```

Comment:
Should this also include a domain in the tag? An operator name is only guaranteed to be unique within a domain afaik. 

e.g.
  ORT_SPECIFY_OP_KERNEL_ARG_ALLOWED_TYPES(ai.onnx, Cast, Input, 0, float, int64_t)

Reply:
added an argument for specifying domain. i don't think it'll take ai.onnx, could we use something like kOnnxDomain?

---
In reply to: [565027440](https://github.com/microsoft/onnxruntime/pull/6466#discussion_r565027440) [](ancestors = 565027440)